### DISCUSSION TITLE
How do I create a new branch in GitHub and switch to it?

### DISCUSSION THREAD

**Discussion by zerox-alpha** [community/community#139484] [New to GitHub]
I'm new to Git and GitHub, and I want to create a new branch for a feature I'm working on. How can I create a new branch from the command line and switch to it in my local repository?

**Comment by im-shubh** [comment]
Creating a new branch and switching to it in Git is quite simple. Here's how you can do it from the command line:

**Create a new branch: Run the following command in your terminal:**
`git checkout -b feature-branch`

**Push the branch to GitHub: After creating the branch locally, you need to push it to your GitHub repository:**
`git push -u origin feature-branch`

**Check your branches: You can check the list of branches and see which one you're currently on with:**
`git branch`

**Reply by zerox-alpha** [reply]
Thanks Got it!

**Comment by ebndev** [comment]
Please see [this comment](https://github.com/orgs/community/discussions/139483#discussioncomment-10730326) for reference to why this discussion was closed and locked. Thanks!
